src/Orleans.Runtime/GrainDirectory/GrainLocatorResolver.cs

```diff
@@ -7,7 +7,7 @@ namespace Orleans.Runtime.GrainDirectory
 {
     internal class GrainLocatorResolver
     {
-        private readonly ConcurrentDictionary<GrainType, IGrainLocator> resolvedLocators = new(GrainType.Comparer.Instance);
+        private readonly ConcurrentDictionary<GrainType, IGrainLocator> resolvedLocators = new();
```

Comment:
What's the reason for this? IIRC, not passing this will cause a lot of `EqualityComparer<T>.Default` lookups, which weren't great for performance. I might be misremembering.

Reply:
That used be the case at some point, but now it's actually better not pass a comparer for value types.